nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1987 @@
+								Optional:    true,
+							},
+							"port": {
+								Type:         schema.TypeInt,
+								Optional:     true,
+								Default:      514,
+								ValidateFunc: validateSinglePort(),
+								Description:  "Syslog server port",
+							},
+							"protocol": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								Description:  "Syslog protocol",
+								Default:      model.SyslogConfiguration_PROTOCOL_UDP,
+								ValidateFunc: validation.StringInSlice(syslogProtocolValues, false),
+							},
+							"server": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Server IP or fqdn",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getEdgeNodeSchema() *schema.Schema {
+	s := map[string]*schema.Schema{
+		"deployment_config": getEdgeNodeDeploymentConfigSchema(),
+		"node_settings":     getEdgeNodeSettingsSchema(),
+	}
+	return getNodeSchema(s, false)
+}
+
+func getHostNodeSchema() *schema.Schema {
+	s := map[string]*schema.Schema{
+		"host_credential": {
+			Type:        schema.TypeList,
+			MaxItems:    1,
+			Description: "Host login credentials",
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"password": {
+						Type:        schema.TypeString,
+						Sensitive:   true,
+						Required:    true,
+						Description: "The authentication password of the host node",
+					},
+					"thumbprint": {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "ESXi thumbprint or SSH key fingerprint of the host node",
+					},
+					"username": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "The username of the account on the host node",
+					},
+				},
+			},
+		},
+		"os_type": {
+			Type:         schema.TypeString,
+			Required:     true,
+			Description:  "Hypervisor OS type",
+			ValidateFunc: validation.StringInSlice(hostNodeOsTypeValues, false),
+		},
+		"os_version": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Hypervisor OS version",
+		},
+		"windows_install_location": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "C:\\Program Files\\VMware\\NSX\\",
+			Description: "Install location of Windows Server on baremetal being managed by NSX",
+		},
+	}
+	return getNodeSchema(s, false)
+}
+
+func getPublicCloudGatewayNodeSchema() *schema.Schema {
+	s := map[string]*schema.Schema{
+		"deployment_config": getEdgeNodeDeploymentConfigSchema(),
+		"node_settings":     getEdgeNodeSettingsSchema(),
+	}
+	return getNodeSchema(s, false)
+}
+
+func getPreconfiguredHostSwitchSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		MaxItems:    1,
+		Optional:    true,
+		Description: "Preconfigured host switch",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"endpoint": {
+					Type:        schema.TypeList,
```

Comment:
Do we need this to be a list?

Reply:
I had the same thought - the fact that the API uses list here is strange and I assumed that it could change to multiple endpoints at some point - and then we'll have to support some legacy. What's your thought about this?

Reply:
Since this API is going to be rewritten with policy, I think it is unlikely it would evolve significantly.

Reply:
Cool so this can be flattened.